jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -57,6 +59,27 @@ public static void premain(String agentArgument, Instrumentation instrumentation
      server.setHandler(context);
      context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
 
-     server.start();
+     final int delayInSecs = collector.getStartDelay();
```

Comment:
The trigger for Collector.collect() is an external request right? So you'll be inserting a fixed delay into retrieving the first set of metrics, even though the container may actually be ready and available.